lighthouse-core/report/html/renderer/base64.js

```diff
@@ -0,0 +1,77 @@
+/**
+ * @license Copyright 2021 The Lighthouse Authors. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+/* global self btoa atob pako CompressionStream Response */
+
+const toBase64 = typeof btoa !== 'undefined' ?
+  btoa :
+  /** @param {string} str */
+  (str) => Buffer.from(str).toString('base64');
+const fromBase64 = typeof atob !== 'undefined' ?
+  atob :
+  /** @param {string} str */
+  (str) => Buffer.from(str, 'base64').toString();
+
+/**
+ * Takes an UTF-8 string and returns a URL-safe base64 encoded string.
+ * If gzip is true, the UTF-8 bytes are gzipped before base64'd, using
+ * CompressionStream (currently only in Chrome), falling back to pako
+ * (which is only used to encode in our Node tests).
+ * @param {string} string
+ * @param {{gzip: boolean}} options
+ * @return {Promise<string>}
+ */
+async function encode(string, options) {
```

Comment:
I do find myself confusing `encode`/`decode` at the callsite, though. What do you think about `stringToBase64`/`base64ToString`?

Reply:
I don't understand the confusion.

Didn't we just say that calling this thing `base64` was bad :)

Reply:
we agreed to not call the *file* `base64` I'm fine with the methods mentioning `base64` in their input/output descriptions (in fact I think this was a [specific request](https://github.com/GoogleChrome/lighthouse/pull/12519#discussion_r637222077) I had)

Reply:
yeah, the `string` part isn't super important, so @patrickhulce's `toBase64`/`fromBase64` suggestion is also good